tss-esapi/src/ffi.rs

```diff
@@ -26,7 +27,60 @@ where
     owned_ffi_data
 }
 
+/// Function that takes ownership of bytes that are stored in a
+/// buffer that has been allocated with C memory allocation functions in TSS.
+///
+/// # Arguments
+/// * `ffi_bytes_ptr` - A pointer to the FFI buffer.
+/// * `size`          - The number of bytes to read from the buffer.
+///
+/// # Returns
+/// The owned bytes in the form of a `Vec<u8>` object.
 pub(crate) fn to_owned_bytes(ffi_bytes_ptr: *mut u8, size: usize) -> Vec<u8> {
     let ffi_bytes = unsafe { MBox::<[u8]>::from_raw_parts(ffi_bytes_ptr, size) };
     return Vec::<u8>::from(ffi_bytes.as_ref());
 }
+
+/// Type used for handling `size_t` variables
+pub(crate) struct FfiSizeType(crate::tss2_esys::size_t);
```

Comment:
I'm wondering - would we ever want to expose anything from this module to the outside world? If not, them maybe we can just tag the entire module `pub(crate)`, then we don't need to worry about having everything in here `pub(crate)` (?)

Reply:
Yes, the FFI module will probably only be used internally.

Reply:
Actually it already is ...
https://github.com/parallaxsecond/rust-tss-esapi/blob/main/tss-esapi/src/lib.rs#L116

Reply:
When some of the ```pub(crate)``` I got some errors about leaking private stuff.